attestation-service/verifier/src/az_tdx_vtpm/mod.rs

```diff
@@ -0,0 +1,144 @@
+// Copyright (c) Microsoft Corporation.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+use super::tdx::claims::generate_parsed_claim;
+use super::tdx::quote::{ecdsa_quote_verification, parse_tdx_quote, Quote as TdQuote};
+use super::{TeeEvidenceParsedClaim, Verifier};
+use crate::{InitDataHash, ReportData};
+use anyhow::{bail, Context, Result};
+use async_trait::async_trait;
+use az_tdx_vtpm::hcl::HclReport;
+use az_tdx_vtpm::vtpm::Quote as TpmQuote;
+use log::{debug, warn};
+use openssl::pkey::PKey;
+use serde::{Deserialize, Serialize};
+
+#[derive(Serialize, Deserialize)]
+struct Evidence {
+    tpm_quote: TpmQuote,
+    hcl_report: Vec<u8>,
+    td_quote: Vec<u8>,
+}
+
+#[derive(Default)]
+pub struct AzTdxVtpm;
+
+#[async_trait]
+impl Verifier for AzTdxVtpm {
+    /// The following verification steps are performed:
+    /// 1. TPM Quote has been signed by AK included in the HCL variable data
+    /// 2. Attestation nonce matches TPM Quote nonce
+    /// 3. TD Quote is genuine
+    /// 4. TD Report's report_data field matches hashed HCL variable data
+    async fn evaluate(
+        &self,
+        evidence: &[u8],
+        expected_report_data: &ReportData,
+        expected_init_data_hash: &InitDataHash,
+    ) -> Result<TeeEvidenceParsedClaim> {
+        let ReportData::Value(expected_report_data) = expected_report_data else {
```

Comment:
```suggestion
        let expected_report_data = regularize_data(expected_report_data, 64, "REPORT_DATA", "Azure TDX vTPM");
        let ReportData::Value(expected_report_data) = expected_report_data else {
```

Please add a sentence before this to prevent from unexpected failure when the given `expected_report_data` is not 64 bytes long. See https://github.com/confidential-containers/kbs/pull/274/files#diff-257710d85707521628ac32349008e86e91a8fe19e49aaa4b16384e125abebe60R62-R63

Reply:
Oh. I once thought that vtpm is also padded to 64bytes.

Reply:
👍 I think it's limited to a maximum of 64, but you can also supply less and the quote will only contain that